dashboard/app/controllers/authentication_options_controller.rb

```diff
@@ -11,8 +11,28 @@ def connect
   def disconnect
     return head(:bad_request) unless current_user&.migrated?
     auth_option = current_user.authentication_options.find(params[:id])
-    auth_option.destroy if auth_option.present?
 
-    return head(:no_content)
+    if auth_option.present?
+      auth_option.destroy
+
+      # Replace primary_contact_info if we just destroyed it
+      if current_user.primary_contact_info_id == auth_option.id
```

Comment:
it looks like we are doing this regardless of whether the user had an email address associated with their account prior to connecting  their account to an auth provider. is this intended? I couldn't tell from the description if `Given I have an email 'x@example.com'` just means that you have an email address, or if it means your code.org account has the email address.

Reply:
Can you help me understand the case you're describing?

If the user has an email associated with their account, it will exist as an `AuthenticationOption` with `credential_type: EMAIL` and `primary_contact_info` will point to it.  If they don't, they won't have a `primary_contact_info` so this check will fail and we won't come up with a replacement.

Reply:
It depends on whether that auth provider is set as the user's primary_contact_info, and is therefore being used to populate their email address.

We're jumping through some hoops with our data model here so that the UI will behave predictably; an action taken in the linked accounts table should not appear to have any effect on the (visually separate) email field, and vice-versa.

Personally I'd prefer that we unified these two separate concepts on the account page, and didn't allow you to delete your "primary" contact info, as GitHub does in their account UI; I pushed for this in the spec but couldn't get the product team to go for it.